hellosign_docsignature_template.go

```diff
@@ -74,17 +74,25 @@ func (m *Client) DeleteTemplate(templateID string) (*http.Response, error) {
 	return response, err
 }
 
-// GetEmbeddedTemplateEditURL - Retrieves an embedded template object.
-// TODO Abhishek: This is an incorrect implementation and should be replaced with a POST call.
-//
-//	A few things need to change and hence leaving it for now.
-func (m *Client) GetEmbeddedTemplateEditURL(templateID string) (*model.EmbeddedTemplateEditURL, error) {
+// GetEmbeddedTemplateEditURL - Retrieves an embedded template object to edit.
+func (m *Client) GetEmbeddedTemplateEditURL(templateID string, customFields string) (*model.EmbeddedTemplateEditURL, error) {
 	if templateID == "" {
 		return nil, fmt.Errorf("invalid argument: %s", templateID)
 	}
+
+	req := model.EditEmbeddedTemplateRequest{}
+	req.ShowPreview = true
```

Comment:
This will always enable the preview mode, which is fine from a user perspective, but every time the user clicks on preview, we will get a callback.

Also I think this is something which we should control by passing it from the service/repo layer and not have it controlled in the SDK.